script/utils/CurrentConfigCheck.s.sol

```diff
@@ -0,0 +1,570 @@
+        payloads[1] = abi.encodeWithSelector(AccessControl.revokeRole.selector, timelockController.EXECUTOR_ROLE(), msg.sender);
+        // 3. remove sender as proposer
+        payloads[2] = abi.encodeWithSelector(AccessControl.revokeRole.selector, timelockController.PROPOSER_ROLE(), msg.sender);
+        // 4. remove sender as admin
+        payloads[3] = abi.encodeWithSelector(AccessControl.revokeRole.selector, timelockController.TIMELOCK_ADMIN_ROLE(), msg.sender);
+
+        // 5. add operationsMultisig as canceller
+        payloads[4] = abi.encodeWithSelector(AccessControl.grantRole.selector, timelockController.CANCELLER_ROLE(), operationsMultisig);
+        // 6. add operationsMultisig as proposer
+        payloads[5] = abi.encodeWithSelector(AccessControl.grantRole.selector, timelockController.PROPOSER_ROLE(), operationsMultisig);
+
+        // 7. add protocolCouncilMultisig as proposer
+        payloads[6] = abi.encodeWithSelector(AccessControl.grantRole.selector, timelockController.PROPOSER_ROLE(), protocolCouncilMultisig);
+        // 8. add protocolCouncilMultisig as executor
+        payloads[7] = abi.encodeWithSelector(AccessControl.grantRole.selector, timelockController.EXECUTOR_ROLE(), protocolCouncilMultisig);
+
+        // 9. add communityMultisig as admin
+        payloads[8] = abi.encodeWithSelector(AccessControl.grantRole.selector, timelockController.TIMELOCK_ADMIN_ROLE(), communityMultisig);
+
+        // TODO: get appropriate value for chain instead of hardcoding
+        uint256 delayToSet;
+        if (timelockController == protocolTimelockController) {
+            delayToSet = 10 days;
+        } else if (timelockController == protocolTimelockController_BEIGEN) {
+            delayToSet = 14 days;
+        }
+        require(delayToSet != 0, "delay not calculated");
+        // 10. set min delay to appropriate length
+        payloads[9] = abi.encodeWithSelector(timelockController.updateDelay.selector, delayToSet);
+
+        // schedule the batch
+        timelockController.scheduleBatch(
+            targets, 
+            values, 
+            payloads, 
+            bytes32(0), // no predecessor needed
+            bytes32(0), // no salt 
+            0 // 0 enforced delay
+        );
+
+        // execute the batch
+        timelockController.executeBatch(
+            targets, 
+            values, 
+            payloads, 
+            bytes32(0), // no predecessor needed
+            bytes32(0) // no salt
+        );
+        vm.stopBroadcast();
+    }
+
+    function simulateLegacyTimelockActions() public {
+        // TODO
+        // give proxy admin ownership to timelock controller
+        // eigenTokenProxyAdmin.transferOwnership(address(timelockController));
+
+        // swapOwner(address previousOwner, address oldOwner, address newOwner)
+        // TODO: figure out if this is the correct input for all chains or the communityMultisig is correct on some
+        address previousOwner = address(1);
```

Comment:
what does this do?

Reply:
This is a required input, and the call will revert without it being correct.
(Gnosis) Safes keep a weird, internal and not-very-easy-to-query linked-list-esque(?) structure of owners which uses address(1) as one end.
Typically this will be another owner of the Safe, but in general as I understand it it's the other node which will need to be attached to the new node to complete the linked structure.

Reply:
ok cool, as long as the test works